src/utils/embedded/embedded.provider.tsx

```diff
@@ -1101,48 +1118,66 @@ export function EmbeddedProvider({ children }: EmbeddedProviderProps) {
         if (activatedWallet) {
           const { id: walletId, address: walletAddress } = activatedWallet;
 
-          const jwk = await WalletUtils.generateWalletJWKFromShares(
-            walletAddress,
-            [activatedWallet.authShare, activatedWallet.deviceShare]
-          );
+          // Check if device share exists for passkey auth
+          const deviceShares = await WalletUtils.getDeviceSharesForUser(userId);
+          const hasDeviceShare = !!deviceShares[walletId];
```

Comment:
Inside this `if`, a wallet has already been activated, so this check should always pass.